src/fastfield/delete.rs

```diff
@@ -38,6 +37,34 @@ pub fn write_delete_bitset(
     Ok(())
 }
 
+/// Merges two `DeleteBitSet` into a new one.
+pub fn merge_delete_bitset(left: &DeleteBitSet, right: &DeleteBitSet) -> DeleteBitSet {
+    let left_data = left.data.as_slice();
+    let right_data = right.data.as_slice();
+
+    let mut merged = vec![];
+    merged.resize(left_data.len().max(right_data.len()), 0);
```

Comment:
len is not what you think it is.
and if it was what you think it is, you are using it wrong.